core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -278,10 +278,11 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
         throw new StateChangeFailedException(failMsg)
       case _ =>
         debug("Live assigned replicas for partition %s are: [%s]".format(topicAndPartition, liveAssignedReplicas))
-        // make the first replica in the list of assigned replicas, the leader
-        val leader = liveAssignedReplicas.head
-        val leaderIsrAndControllerEpoch = new LeaderIsrAndControllerEpoch(new LeaderAndIsr(leader, liveAssignedReplicas.toList),
-          controller.epoch)
+        val leaderIsrAndControllerEpoch = LeaderIsrAndControllerEpoch(
+          // leader is the first replica in the list of assigned replicas
+          LeaderAndIsr(liveAssignedReplicas.head, liveAssignedReplicas.toList),
```

Comment:
Agreed about the existing code. Actually I did such a improvements in the previous PR, but @guozhangwang proposed to do smaller changes in many PRs.
Generally speaking, it is difficult for me to know when can I rewrite other code which appears in the same file. Will do that in this case.